c++/src/capnp/compat/http-over-capnp.c++

```diff
@@ -688,6 +825,80 @@ public:
     });
   }
 
+  kj::Promise<void> connect(ConnectContext context) override {
+    auto params = context.getParams();
+    auto host = params.getHost();
+    auto headers = factory.headersToKj(params.getHeaders());
+    auto upPipe = kj::newOneWayPipe();
+
+    PipelineBuilder<ConnectResults> pb;
+    pb.setUp(factory.streamFactory.kjToCapnp(kj::mv(upPipe.out)));
+
+    context.setPipeline(pb.build());
+
+    // This is similar to kj::HttpClientAdapter::connect.
+    auto pipe = kj::newTwoWayPipe();
+    auto paf = kj::newPromiseAndFulfiller<kj::HttpClient::ConnectRequest::Status>();
+    auto response = kj::refcounted<HttpOverCapnpConnectResponseImpl>(kj::mv(paf.fulfiller));
+
+    auto promise = inner->connect(host, headers, *pipe.ends[0], *response)
+        .eagerlyEvaluate([response=kj::mv(response),
+                          host=kj::mv(host),
+                          headers=kj::mv(headers),
+                          connectStream=kj::mv(pipe.ends[0])](kj::Exception&& ex) mutable {
+        response->handleException(kj::mv(ex), kj::mv(connectStream));
+      });
+
+    return paf.promise.then(
+        [this, upPipe = kj::mv(upPipe), down = params.getDown(), context = kj::mv(context),
+        io = kj::mv(pipe.ends[1])](auto status) mutable {
+      auto headers = status.headers->clone();
+
+      if (status.statusCode < 200 || status.statusCode > 299) {
```

Comment:
Maybe this should be `KJ_IF_MAYBE(errorBody, status.errorBody) {`? Seems a bit safer than relying on status code conventions, plus you can avoid the check later.

Reply:
Isn't it possible to get no error body but still have non 2xx status code?